### DISCUSSION TITLE
Customize source path of Emacs

### DISCUSSION THREAD

**Discussion by gagbo** [d12frosted/homebrew-emacs-plus#523] [Q&A] [answered]
Hello,

First and foremost thanks for maintaining the recipe :)

I was wondering if it is possible to customize the source of Emacs in the recipe, just like nix does with overlays or how the package definitions can be overridden in guix.

Basically I'd like to use [emacs-lsp fork](https://github.com/emacs-lsp/emacs) which runs the json-rpc I/O tasks in a separate thread from UI (might be a bad summary but it's close enough if you're curious)

Cheers,
Gerry

**Comment by d12frosted** [comment, marked as answer]
Hey @gagbo

AFAIK, brew doesn't allow this. The only option is to `brew edit FORMULA` and that's it. In this regards nix is more powerful :) 

It's also worth mentioning, that this feature sounds interesting, so we can either (a) add an option to build from this fork or (b) extract a patch and apply it under extra option.

**Reply by gagbo** [reply]
It's still a little bit experimental (i.e. it's mostly in a public beta state), so I'd rather jsut test things locally before trying to formally add the option to the formula. I'll play with `brew edit emacs-plus` for the time being then!

**Reply by d12frosted** [reply]
Ok, let me know if you need help with that.

**Reply by gagbo** [reply]
It's okay for the time being. I got something in a decent enough state so that I can test and give pointers to yyoncho. The fork is definitely not in a finished/publishable test though, so I don't advise trying to run this unless you want to give feedback and debug a little bit https://github.com/d12frosted/homebrew-emacs-plus/compare/master...gagbo:homebrew-emacs-plus:emacs-lsp?expand=1
